From a3d1082ce648b46ea79d4e37faf0bb0feab3b158 Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Tue, 15 Jun 2021 11:31:49 +0200
Subject: [PATCH 6/6] tools: generate human readable output in support bundles

str() will just output the undecoded string with e.g. quoted newlines,
which does not make really readable output.

To get actual output we need to decode() the output, which also makes
the str() call redundant.

Before:

  >>[2021-06-15 09:28:12.474269]show zebra
  b'                            Route      Route      Neighbor   LSP        LSP\nVRF                         Installs   Removals    Updates   Installs   Removals\ndefault                          249         20          0          0          0\n'########################################################

After:

  >>[2021-06-15 09:26:11.778352]show zebra
                              Route      Route      Neighbor   LSP        LSP
  VRF                         Installs   Removals    Updates   Installs   Removals
  default                          249         20          0          0          0
  ########################################################

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 tools/generate_support_bundle.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/generate_support_bundle.py b/tools/generate_support_bundle.py
index a414dc12c..450f2f861 100755
--- a/tools/generate_support_bundle.py
+++ b/tools/generate_support_bundle.py
@@ -56,7 +56,7 @@ def executeCommand(cmd, outputFile):
         try:
             dateTime = datetime.datetime.now()
             outputFile.write(">>[" + str(dateTime) + "]" + cmd + "\n")
-            outputFile.write(str(cmd_output))
+            outputFile.write(cmd_output.decode())
             outputFile.write(
                 "########################################################\n"
             )
@@ -66,10 +66,10 @@ def executeCommand(cmd, outputFile):
     except subprocess.CalledProcessError as e:
         dateTime = datetime.datetime.now()
         outputFile.write(">>[" + str(dateTime) + "]" + cmd + "\n")
-        outputFile.write(str(e.output))
+        outputFile.write(e.output.decode())
         outputFile.write("########################################################\n")
         outputFile.write("\n")
-        print("Error:" + str(e.output))
+        print("Error:" + e.output.decode())
 
 
 # Process the support bundle configuration file
-- 
2.32.0

